### PR TITLE
Some consistency fixes for api/subjects/{login}/sources

### PR DESCRIPTION
**Pull request by blootsvoets**

### CODE REVIEW

File: src/main/java/org/radarcns/management/service/mapper/SourceMapper.java [hunk 1]
```diff
@@ -39,11 +39,13 @@
 
     @Mapping(source = "sourceId" , target = "sourceId")
     @Mapping(source = "expectedSourceName" , target = "expectedSourceName")
+    @Mapping(source = "deviceType.id" , target = "deviceTypeId")
     @Mapping(source = "deviceType.deviceModel" , target = "deviceTypeModel")
     @Mapping(source = "deviceType.deviceProducer" , target = "deviceTypeProducer")
     @Mapping(source = "deviceType.catalogVersion" , target = "deviceCatalogVersion")
     @Mapping(source = "project.id" , target = "projectId")
     @Mapping(source = "assigned" , target = "assigned")
```
Comment (nivemaham) [inline src/main/java/org/radarcns/management/service/mapper/SourceMapper.java]: Mapping of same attributes will work with or without Mapping annotation.

**Comment by blootsvoets** [comment]
Yes, good point. I'll take a stab at that.

File: src/main/java/org/radarcns/management/service/mapper/SourceMapper.java (hunk 1 above)
Comment (blootsvoets) [reply src/main/java/org/radarcns/management/service/mapper/SourceMapper.java]: Thanks, I wasn't aware

File: src/main/java/org/radarcns/management/service/dto/SubjectDTO.java [hunk 2]
```diff
@@ -44,7 +46,7 @@
 
     private Set<MinimalSourceDetailsDTO> sources = new HashSet<>();
 
-    private Set<AttributeMapDTO> attributes = new HashSet<>();
+    private Map<String, String> attributes = new HashMap<>();
```
[the remark this answers is not in the document]
Comment (blootsvoets) [reply src/main/java/org/radarcns/management/service/dto/SubjectDTO.java]: Ok, created a Javascript `DictionaryMapperComponent` to use a map of `[key: string]: string` instead. This also solves some uniqueness and interaction bugs.

**Comment by nivemaham** [comment]
looks good. I think later on we can convert projectDTO also to use Map directly. Good solution @blootsvoets

**Pull request merged**